src/dispatching/dialogue/storage/sqlite_storage.rs

```diff
@@ -60,16 +59,21 @@ where
 {
     type Error = SqliteStorageError<<S as Serializer<D>>::Error>;
 
+    /// Returns [`sqlx::Error::RowNotFound`] if a dialogue does not exist.
     fn remove_dialogue(
         self: Arc<Self>,
         chat_id: i64,
     ) -> BoxFuture<'static, Result<(), Self::Error>> {
         Box::pin(async move {
-            if get_dialogue(&self.pool, chat_id).await?.is_some() {
-                sqlx::query("DELETE FROM teloxide_dialogues WHERE chat_id = ?")
+            let deleted_rows_count =
+                sqlx::query("DELETE FROM teloxide_dialogues WHERE chat_id = ?; SELECT changes()")
```

Comment:
Is the `SELECT changes()` required for `.rows_affected()`?

Reply:
Nope, fixed.